core/src/metadata_json.rs

```diff
@@ -65,6 +67,25 @@ pub struct MetadataJsonDownloadWorkerArgs {
     metadata_json_download_worker_request_timeout: u64,
 }
 
+async fn is_asset_data_fetch_req(
+    download_metadata_info: &DownloadMetadataInfo,
+    pool: Pool<Postgres>,
+) -> bool {
+    let DownloadMetadataInfo {
+        asset_data_id,
+        slot: incoming_slot,
+        ..
+    } = download_metadata_info;
+
+    let conn = SqlxPostgresConnector::from_sqlx_postgres_pool(pool);
+
+    asset_data::Entity::find_by_id(asset_data_id.clone())
+        .one(&conn)
+        .await
+        .unwrap_or(None)
+        .is_some_and(|model| *incoming_slot > model.slot_updated)
```

Comment:
The check is that the incoming url and slot < the incoming.

The other item to audit is the usage of reindex. These should be aligned so that this is done before they are marked for reindex and then just go off of it. 

Can it get in a spot where it never reaches the metaplex json downloader unless it needs to be downloaded?

Reply:
this can be done using reindex flag , we only call download_metadata_notifier if reindex flag is set  . like here ?                                         
![image](https://github.com/user-attachments/assets/7fc646d9-177b-4b42-a6fa-c80971c21a3e)